qiskit_optimization/algorithms/goemans_williamson_optimizer.py

```diff
@@ -0,0 +1,254 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Implementation of the Goemans-Williamson algorithm as an optimizer.
+Requires CVXPY to run.
+"""
+
+from typing import Optional, List, Tuple, Union
+
+import numpy as np
+from qiskit.exceptions import MissingOptionalLibraryError
+
+from qiskit_optimization import QuadraticProgram
+from qiskit_optimization.algorithms import OptimizationResult, OptimizationResultStatus, \
+    OptimizationAlgorithm, SolutionSample
+from qiskit_optimization.applications.ising.max_cut import max_cut_value
+from qiskit_optimization.problems import Variable
+
+try:
+    import cvxpy as cvx
+    _HAS_CVXPY = True
+except ImportError:
+    _HAS_CVXPY = False
+
+
+class GoemansWilliamsonOptimizationResult(OptimizationResult):
```

Comment:
Added all new classes, hopefully in the correct place/order.

Reply:
Seems fine thx. We will need to review the documentation ahead of release anyway. Given that the folders like algorithms etc are now toplevel the main page ought to be altered to reflect/enumerate these directly rather than via submodules. If there is some better grouping of the algorithms in the documentation that makes more sense for users that can be done then too.